File: examples/check-console-errors.mjs
/**
 * Webcompat repro pattern: JavaScript errors on page load
 *
 * Console events are not captured by BiDiPage — subscribe via the connection directly.
 * Replace the URL with the site under investigation.
 *
 * Firefox: ff-bidi-run examples/check-console-errors.mjs --headless
 * Chrome:  start chromium-bidi or chromedriver first, then:
 *          ff-bidi-run examples/check-console-errors.mjs --browser chrome --bidi-url ws://localhost:9222/session
 */
import { startSession } from 'ff-test-firefox-bidi-client';

const TARGET_URL = 'https://example.com';

const session = await startSession();
const { page, connection } = session;

const errors = [];
connection.onEvent('log.entryAdded', params => {
  if (params.level === 'error') errors.push(params.text);
});
await connection.subscribe(['log.entryAdded']);

try {
  await page.goto(TARGET_URL);

  // Wait briefly for any async scripts to fire errors
  await new Promise(r => setTimeout(r, 1000));

  if (errors.length > 0) {
    console.log('issue reproduced');
    for (const text of errors) {
      console.log(`  [error] ${text}`);
    }
  } else {
    console.log('issue not reproduced');
  }
} finally {
  await session.close();
}

File: python/examples/check_console_errors.py
"""
Webcompat repro pattern: JavaScript errors on page load

Run:
    python check_console_errors.py
    FIREFOX_PATH=/path/to/firefox FIREFOX_HEADLESS=1 python check_console_errors.py
"""

import asyncio
from firefox_bidi_client import start_session

TARGET_URL = "https://example.com"


async def main():
    async with await start_session() as session:
        page, connection = session.page, session.connection

        errors = []
        connection.on_event("log.entryAdded", lambda p: errors.append(p["text"])
                            if p.get("level") == "error" else None)
        await connection.subscribe(["log.entryAdded"])

        await page.goto(TARGET_URL)

        # Wait briefly for async scripts to fire errors
        await asyncio.sleep(1)

        if errors:
            print("issue reproduced")
            for text in errors:
                print(f"  [error] {text}")
        else:
            print("issue not reproduced")


asyncio.run(main())
